spec/acceptance/windows_spec.rb

```diff
@@ -2,69 +2,57 @@
+      result = run_bolt_task('package::windows', 'action' => 'install', 'name' => package_to_use)
+      expect(result.exit_code).to eq(0)
+      expect(result.stdout).to match(%r{Install})
     end
   end
 
   describe 'uninstall action' do
     it "uninstall #{package_to_use}" do
-      apply_manifest_on(default, "package { \"#{package_to_use}\": ensure => present, }")
-      result = run_task('package::windows', 'default', 'action' => 'uninstall', 'name' => package_to_use)
-      expect(result[0]).to include('status' => 'success')
-      expect(result[0]['result']).to include('status' => %r{Uninstall})
+      apply_manifest("package { \"#{package_to_use}\": ensure => present, provider => 'chocolatey' }")
+      result = run_bolt_task('package::windows', 'action' => 'uninstall', 'name' => package_to_use)
+      expect(result.exit_code).to eq(0)
+      expect(result.stdout).to match(%r{Uninstall})
     end
   end
 
   describe 'install specific' do
     it 'upgrade notepad++ to a specific version' do
-      result = run_task('package::windows', 'default', 'action' => 'upgrade', 'name' => package_to_use, 'version' => '7.5.5')
-      expect(result[0]).to include('status' => 'success')
-      expect(result[0]['result']).to include('status' => %r{Upgrade})
-      expect(result[0]['result']).to include('old_version')
-      expect(result[0]['result']).to include('version')
+      result = run_bolt_task('package::windows', 'action' => 'upgrade', 'name' => package_to_use, 'version' => '7.5.5')
+      expect(result.exit_code).to eq(0)
+      expect(result.stdout).to match(%r{Upgrade})
     end
 
     it 'upgrade notepad++' do
-      result = run_task('package::windows', 'default', 'action' => 'upgrade', 'name' => package_to_use)
-      expect(result[0]).to include('status' => 'success')
-      expect(result[0]['result']).to include('status' => %r{Upgrade})
-      expect(result[0]['result']).to include('old_version')
-      expect(result[0]['result']).to include('version')
+      result = run_bolt_task('package::windows', 'action' => 'upgrade', 'name' => package_to_use)
+      expect(result.exit_code).to eq(0)
+      expect(result.stdout).to match(%r{Upgrade})
     end
   end
 
   context 'when puppet-agent feature not available on target' do
-    let(:bolt_inventory) { hosts_to_inventory }
-
     it 'status action fails' do
+      inventory_hash = inventory_hash_from_inventory_file
+      inventory_hash = remove_feature_from_group(inventory_hash, 'puppet-agent', 'winrm_nodes')
+      write_to_inventory_file(inventory_hash, 'inventory.yaml')
       params = { 'action' => 'status', 'name' => package_to_use }
-      result = run_task('package', 'default', params)
-      expect(result[0]).to include('status' => 'failure')
-      expect(result[0]['result']).to include('status' => 'failure')
-      expect(result[0]['result']['_error']).to include('msg' => %r{'status' action not supported})
```

Comment:
Really exciting to see this being ported to litmus! Without having seen how `Error` results are returned form `run_bolt_task` (perhaps we need like an `run_failed_bolt_task` or something), I think that it will be important to ensure that these particular errors are surfaced. Specifically the case where the init implementation is specified on CLI, but the absence of the `puppet-agent` feature causes an alternate implementation to be executed.

Reply:
we could use something similar to https://github.com/puppetlabs/puppet_litmus/blob/bb60e69143e21252c6b5652498afeaa8eac9cf43/lib/puppet_litmus/serverspec.rb#L96 ie opts[:expect_failures]

Reply:
https://github.com/puppetlabs/puppet_litmus/pull/146/files